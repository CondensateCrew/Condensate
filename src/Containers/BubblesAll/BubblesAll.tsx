import React, {useState, useEffect} from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { AppStore, WordSample } from 'interfaces';
import { Redirect } from 'react-router-dom';
// import collection from './mockData';
import { addChosenWord } from 'redux/actions';

const BubblesAll: React.FC = () => {
  const dispatch = useDispatch();
  const { timeEnded, randomWordCollections } = useSelector((store: AppStore) => ({
    timeEnded: store.timeEnded,
    randomWordCollections: store.randomWordCollections
  }));

  let randomWords = randomWordCollections.map((word: WordSample) => word.word)
  const firstStack = randomWords.splice(0, 8);

  const [ bubbles, setBubbles ] = useState<string[]>(firstStack);

  useEffect(() => {
    const timer = setTimeout(() => {
      setBubbles(randomWords.splice(0, 8));
    }, 8000);

    return () => clearTimeout(timer);
  }, []); //eslint-disable-line

  const sendToChosen = (event: React.MouseEvent) => {
    const currentTarget = event.target as HTMLElement;
    updateElements(currentTarget);
    dispatch(addChosenWord(currentTarget.id));
  }

  const bubbleEls = bubbles.map((bubble: string, i: number) => (
    <p
      key={`${bubble}-${i + 1}`}
      className="bubble randomly">
      <span></span>
      <button id={bubble.toLowerCase()} onClick={sendToChosen}>{bubble}</button>
    </p>
  ));

  return (
    <section className="all-bbls">
      { !timeEnded
        ? bubbleEls
        : <Redirect to='/instructions/second-rd' />
      }
    </section>
  )
}

export const updateElements = (currentTarget: HTMLElement): void => {
  const parent = currentTarget.parentNode as HTMLElement;
  const sibling = currentTarget.previousSibling as HTMLElement;
  parent.classList.add('disappeared');
  sibling.classList.add('bubble-rounds');
  currentTarget.style.animation = 'disappearing 10s';
}

export default BubblesAll;
