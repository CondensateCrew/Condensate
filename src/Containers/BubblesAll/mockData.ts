import { WordSample } from 'interfaces';

// const collection: WordSample = {
//   'container': {
//     type: 'noun',
//     connectedWord: ['package', 'vessel', 'box']
//   },
//   'book': {
//     type: 'noun',
//     connectedWord: ['worm', 'knowledge', 'library']
//   },
//   'sport': {
//     type: 'noun',
//     connectedWord: ['player', 'ball', 'point']
//   },
//   'time': {
//     type: 'noun',
//     connectedWord: ['clock', 'countdown', 'space']
//   },
//   'cloud': {
//     type: 'noun',
//     connectedWord: ['infrastructure', 'network', 'remote']
//   },
//   'water': {
//     type: 'noun',
//     connectedWord: ['compounds', 'hydro', 'erosion']
//   },
//   'security': {
//     type: 'noun',
//     connectedWord: ['safe', 'protection', 'hacker']
//   },
//   'color': {
//     type: 'noun',
//     connectedWord: ['hue', 'shade', 'luminosity']
//   },
//   'software': {
//     type: 'noun',
//     connectedWord: ['codebase', 'application', 'test']
//   },
//   'school': {
//     type: 'noun',
//     connectedWord: ['institution', 'board', 'classroom']
//   },
//   'goat': {
//     type: 'noun',
//     connectedWord: ['animal', 'mountain', 'hoof']
//   },
//   'sunset': {
//     type: 'noun',
//     connectedWord: ['day', 'twilight', 'setting']
//   },
//   'lesson': {
//     type: 'noun',
//     connectedWord: ['assessment', 'instruction', 'material']
//   },
//   'goose': {
//     type: 'noun',
//     connectedWord: ['dinner', 'farm', 'egg']
//   },
//   'park': {
//     type: 'noun',
//     connectedWord: ['lake', 'wood', 'walker']
//   },
//   'door': {
//     type: 'noun',
//     connectedWord: ['entrance', 'exit', 'gateway']
//   },
// };

// export default collection;
