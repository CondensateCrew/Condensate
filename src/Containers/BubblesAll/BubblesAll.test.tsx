import React from 'react';
import BubblesAll from './BubblesAll';
import { shallow } from 'enzyme';
const mockDispatch = jest.fn();

let mockTime: boolean;

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => ({
    timeEnded: mockTime,
    randomWordCollections: [{
      word: 'Goose',
      sentence: 'Goose is walking'
    }]
  })
}));

describe('BubblesAll component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    mockTime = false;
    const wrapper = shallow(
      <BubblesAll />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if time is ended', () => {
    mockTime = true;
    const wrapper = shallow(
      <BubblesAll />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
