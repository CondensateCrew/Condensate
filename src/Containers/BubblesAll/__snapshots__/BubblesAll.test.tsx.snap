// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BubblesAll component should match the snapshot 1`] = `
<section
  className="all-bbls"
>
  <p
    className="bubble randomly"
    key="Goose-1"
  >
    <span />
    <button
      id="goose"
      onClick={[Function]}
    >
      Goose
    </button>
  </p>
</section>
`;

exports[`BubblesAll component should match the snapshot if time is ended 1`] = `
<section
  className="all-bbls"
>
  <Redirect
    to="/instructions/second-rd"
  />
</section>
`;
